plugins/vpp/ifplugin/descriptor/interface_crud.go

```diff
@@ -316,6 +317,16 @@ func (d *InterfaceDescriptor) Delete(key string, intf *interfaces.Interface, met
 func (d *InterfaceDescriptor) Update(key string, oldIntf, newIntf *interfaces.Interface, oldMetadata *ifaceidx.IfaceMetadata) (newMetadata *ifaceidx.IfaceMetadata, err error) {
 	ifIdx := oldMetadata.SwIfIndex
 
+	// af-packet interface with changed mac address has to be re-created
+	if oldIntf.GetType() == interfaces.Interface_AF_PACKET && oldIntf.PhysAddress != newIntf.PhysAddress {
```

Comment:
Why is this part of `Update` and not as `ModifyWithRecreate` so scheduler can handle the calls?

Reply:
Good point, I forgot about that option

Reply:
It might cause issues otherwise. The scheduler has to recreate anything dependent on AFPacket on recreation and this way it would not know it was recreated.